src/main/scala/Engine.scala

```diff
@@ -59,16 +65,41 @@ case class DateRange(
     after: Option[String]) // both empty should be ignored
   extends Serializable
 
+
+/********************************************************************************************************/
+// This is for price range filter
+case class PriceRange(
+                       name: String, // name of item property for the date comparison
+                       lessthan: Option[String], // empty strings means no filter
+                       greaterthan: Option[String]) // both empty should be ignored
+  extends Serializable
+/*********************************************************************************************************/
+
 /** results of a MMRAlgoritm.predict */
 case class PredictedResult(
-    itemScores: Array[ItemScore])
+                            itemScores: Array[ItemScore],
+                            includeItems : Array[IncludeItem])
   extends Serializable
```

Comment:
this is not correct indentation by the Scala style guide. http://docs.scala-lang.org/style/indentation.html#methods_with_numerous_arguments